metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -116,6 +122,9 @@ public void execute(Tuple tuple) {
       boolean ackTuple = !writer.handleAck();
       int numWritten = 0;
       if(sensorParserConfig != null) {
+        if (configUpdatedFlag.getAndSet(false)) {
+          parser.configurationUpdated(getSensorParserConfig().getParserConfig());
```

Comment:
nice catch on the need to re-call getSensorParserConfig().  But I think it needs to be captured in the sensorParserConfig variable for use in lines 132 and 134, eg:
        if (configUpdatedFlag.getAndSet(false)) {
          sensorParserConfig = getSensorParserConfig();  //re-fetch in case it's changed
          parser.configurationUpdated(sensorParserConfig.getParserConfig());
        }